store/postgres/src/entities.rs

```diff
@@ -677,20 +697,72 @@ impl Table {
     }
 
     fn count_entities(&self, conn: &PgConnection) -> Result<u64, Error> {
-        match self {
+        let (count, subgraph) = match self {
             Table::Public(subgraph) => {
                 let count: i64 = public::entities::table
                     .filter(public::entities::subgraph.eq(subgraph.to_string()))
                     .count()
                     .get_result(conn)?;
-                Ok(count as u64)
+                (count as u64, subgraph)
             }
             Table::Split(entities) => {
                 let table = entities.table.clone();
                 let count: i64 = table.count().get_result(conn)?;
-                Ok(count as u64)
+                (count as u64, &entities.subgraph)
             }
-        }
+        };
+        let query = format!(
```

Comment:
This is one nasty side effect. I'd rather not do it this way. Instead, I would move the side effect of setting the `entityCount` into the store if there is no `entityCount` yet.

I get the idea of transitioning subgraphs over to have `entityCount` set incrementally, but I'd prefer for the "get and update" logic to live in the store.

Reply:
It could live even higher, in the GraphQL resolver, because at the time an `entityCount` field value exists on the deployment entity, the resolver doesn't even have to go to the store anymore.

So the `StoreResolver` could call `compute_and_return_entity_count` or similar if `entityCount` is absent and otherwise just return its value.

Reply:
My problem with doing it this way is twofold: (1) it adds writes to a code path that should really only be reads (2) the subgraph writer and this code could race against each other, and if the writer loses the race, we have an incorrect count in there.

I think we have no choice on the query side but to run `select count(*) ...` if the `entityCount` isn't set yet, and only initialize it in the code that writes to the subgraph anyway. The other option with initialization would be to do it in a migration, or on subgraph start, but that would mean that startup would be very slow when this gets rolled out. Though thinking about it, we probably should initialize it in a migration because with ENS, a few queries that kick off a `select count(*) ..` are catastrophic for the server.